src/contracts/Protocol/BountyQuestion.sol

```diff
@@ -9,63 +9,40 @@ import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/utils/Counters.sol";
 import "./modifiers/OnlyAPI.sol";
 
-// TODO remove ERC721 stuff here
-// TODO introduce a Challenge data model
-// Question -> Challenge is many -> one mapping
-// auto-transition from question -> challenge on:
-//   Voting threshold -> do we want manual approval? -> yes
-//   Admin approval (Messari group wants to write their own challenges)
-// TODO ability for author to unlock and "kill" their question -> reimburse voters
-
-// philosphy -> start out gated and the protocol can evolve in the same way the dao does
-
 /// @custom:security-contact contracts@metricsdao.xyz
-contract BountyQuestion is ERC721, ERC721Enumerable, ERC721URIStorage, ERC721Burnable, Ownable, OnlyApi {
+contract BountyQuestion is Ownable, OnlyApi {
     using Counters for Counters.Counter;
 
     Counters.Counter private _tokenIdCounter;
-    mapping(address => QuestionData[]) public authors; // TODO if we want questions to be transferable, then owner != author
-    mapping(uint256 => uint256) private _createdAt;
 
-    constructor() ERC721("MetricsDAO Question", "MDQ") {
+    // This maps the author to the list of token IDs they have created
+    mapping(address => uint256[]) public authors;
+
+    // This maps the token ID to the question data
+    mapping(uint256 => QuestionData) public questions;
+
+    constructor() {
         _tokenIdCounter.increment();
     }
 
-    // working standard metadata format:  Title, Description, Program
-    function safeMint(address to, string calldata uri) public onlyApi returns (uint256) {
+    function createQuestion(address author, string calldata uri) public onlyApi returns (uint256) {
         uint256 tokenId = _tokenIdCounter.current();
         _tokenIdCounter.increment();
-        _safeMint(to, tokenId);
-        _setTokenURI(tokenId, uri);
-        QuestionData memory _question = QuestionData({tokenId: tokenId, url: uri});
-        authors[to].push(_question);
-        return tokenId;
-    }
 
-    //------------------------------------------------------ Solidity Overrides
-
-    function _beforeTokenTransfer(
-        address from,
-        address to,
-        uint256 tokenId
-    ) internal override(ERC721, ERC721Enumerable) {
-        super._beforeTokenTransfer(from, to, tokenId);
-    }
-
-    function _burn(uint256 tokenId) internal override(ERC721, ERC721URIStorage) {
-        super._burn(tokenId);
-    }
-
-    function tokenURI(uint256 tokenId) public view override(ERC721, ERC721URIStorage) returns (string memory) {
-        return super.tokenURI(tokenId);
-    }
-
-    function supportsInterface(bytes4 interfaceId) public view override(ERC721, ERC721Enumerable) returns (bool) {
-        return super.supportsInterface(interfaceId);
+        questions[tokenId] = QuestionData({tokenId: tokenId, url: uri});
+        // QuestionData memory _question = QuestionData({tokenId: tokenId, url: uri});
+        authors[author].push(tokenId);
```

Comment:
I think I get it but if I query by author I should be able to get the tokenId/questionId and also the url associated with it that's why I made those changes - we need that connection

Reply:
I think this is covered by the view function below (and the discussion)